### PR TITLE
Added annotation to exclude from code coverage

### PR DESCRIPTION
**Pull request by megheshpatil**
# Description

- Added Annotation to exclude from code coverage.
- Increased the code coverage

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by whisper6284 — APPROVED** [review]

File: sdk/communication/azure-communication-phonenumbers/pom.xml [hunk 1]
```diff
@@ -170,6 +170,19 @@
           </rules>
         </configuration>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-javadoc-plugin</artifactId>
```
Comment (srnagar) [inline sdk/communication/azure-communication-phonenumbers/pom.xml:175]: Why do we have this plugin overriding the configuration in the parent pom?

File: sdk/communication/azure-communication-phonenumbers/src/main/java/com/azure/communication/phonenumbers/CodeCoverageAnnotation.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import com.azure.communication.phonenumbers.CodeCoverageAnnotation.Generated;
+
+/** Class usedfor code coverage. */
+@Generated
+public class CodeCoverageAnnotation {
```
Comment (srnagar) [inline sdk/communication/azure-communication-phonenumbers/src/main/java/com/azure/communication/phonenumbers/CodeCoverageAnnotation.java:16]: Why is this class needed?

Comment (megheshpatil) [reply sdk/communication/azure-communication-phonenumbers/src/main/java/com/azure/communication/phonenumbers/CodeCoverageAnnotation.java:16]: To add an annotation that will exclude classes from code coverage

File: sdk/communication/azure-communication-phonenumbers/src/main/java/com/azure/communication/phonenumbers/CodeCoverageAnnotation.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import com.azure.communication.phonenumbers.CodeCoverageAnnotation.Generated;
+
+/** Class used for code coverage. */
+@Generated
+public class CodeCoverageAnnotation {
```
Comment (whisper6284) [inline sdk/communication/azure-communication-phonenumbers/src/main/java/com/azure/communication/phonenumbers/CodeCoverageAnnotation.java:16]: How is this file generated? It says that it is code generated, which means it should be in a /generated file/

**Pull request merged**